Source/SwiftLintBuiltInRules/Rules/Lint/UnneededSynthesizedInitializerRule.swift

```diff
@@ -0,0 +1,312 @@
+// swiftlint:disable file_header
+//
+// Adapted from swift-format's UseSynthesizedInitializer.swift
+//
+// https://github.com/apple/swift-format
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+
+import SwiftSyntax
+
+struct UnneededSynthesizedInitializerRule: SwiftSyntaxCorrectableRule, ConfigurationProviderRule {
+    var configuration = SeverityConfiguration(.warning)
+
+    static let description = RuleDescription(
+        identifier: "unneeded_synthesized_initializer",
+        name: "Unneeded Synthesized Initializer",
+        description: "This initializer would be synthesized automatically - you do not need to define it",
+        kind: .lint,
```

Comment:
Not very important, but for me it seems that this rule would better fit into the `.idiomatic` category.